### PR TITLE
add last_activity_at to keep_to_library and keep_to_user

### PR DESCRIPTION
**Pull request by camhashemi**
@ryanpbrewster @leogrim @andrewconner 

we need this to sort by `lastActivityAt` on the feed without doing joins between `bookmark` and `keep_to_user` or `keep_to_library`.

would do the migration tonight, estimated 25 min+ per table given @ryanpbrewster 's migration on `bookmark`

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/model/KeepToLibrary.scala [hunk 1]
```diff
@@ -29,6 +30,9 @@ case class KeepToLibrary(
   def withAddedBy(newOwnerId: Id[User]): KeepToLibrary = this.copy(addedBy = newOwnerId)
   def withUriId(newUriId: Id[NormalizedURI]) = this.copy(uriId = newUriId)
 
+  // denormalized from Keep.lastActivityAt, use in KeepCommander.updateLastActivityAtIfLater
+  def withLastActivityAt(time: DateTime): KeepToLibrary = this.copy(lastActivityAt = time)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/model/KeepToLibrary.scala:34]: Sometimes I like putting the "if later" logic here, so

```
def withActivityAt(time: DateTime) = if (lastActivityAt isAfter time) this else this.copy(lastActivityAt = time)
```

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminBookmarkController.scala [hunk 2]
```diff
@@ -326,4 +328,30 @@ class AdminBookmarksController @Inject() (
...
+
+    model match {
+      case "ktu" =>
+        val ktus = db.readOnlyMaster(implicit s => ktuRepo.allActive)
+        ChunkedResponseHelper.chunked(ktus) { ktu =>
+          db.readWrite { implicit s =>
+            val keep = keepRepo.get(ktu.keepId)
+            ktuRepo.save(ktu.copy(lastActivityAt = keep.lastActivityAt))
+            Json.stringify(Json.obj("ktuId" -> ktu.id.get))
+          }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminBookmarkController.scala]: This db session is probably going to stay open too long. Maybe chunk it?

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/DiscussionCommander.scala [hunk 3]
```diff
@@ -101,7 +101,7 @@ class DiscussionCommanderImpl @Inject() (
       val keep = db.readWrite { implicit s =>
         val oldKeep = keepRepo.get(keepId)
         val allUsers = oldKeep.connections.users ++ newUsers + userId
-        keepCommander.persistKeep(oldKeep.withParticipants(allUsers))
+        keepCommander.addUsersToKeep(keepId, addedBy = userId, allUsers)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/DiscussionCommander.scala:104]: :+1:

File: kifi-backend/modules/common/app/com/keepit/model/KeepToLibrary.scala (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/common/app/com/keepit/model/KeepToLibrary.scala:34]: It's reasonable, but this method shouldn't get any clever ideas, it should just copy straight from `keep.lastActivityAt`, no matter what.

Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/model/KeepToLibrary.scala:34]: Yeah, especially given that you're going for explicit commander methods that do these things I think your way is easier to understand.

**Comment by camhashemi** [comment]
the query was super easy to refactor, tested in dev mode, and things are working as expected. after the migration tonight, I'm going to deploy a separate stream endpoint to test with and hopefully that goes well.

**Pull request merged**